shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java

```diff
@@ -73,18 +82,22 @@ public void onStateUpdate(SplitInstallSessionState state) {
                   TAG,
                   String.format(
                       "Module \"%s\" (sessionId %d) install successfully.",
-                      sessionIdToName.get(state.sessionId()), state.sessionId()));
-              loadAssets(
-                  sessionIdToLoadingUnitId.get(state.sessionId()),
-                  sessionIdToName.get(state.sessionId()));
+                      sessionIdToName.get(sessionId), sessionId));
+              loadAssets(sessionIdToLoadingUnitId.get(sessionId), sessionIdToName.get(sessionId));
               // We only load Dart shared lib for the loading unit id requested. Other loading units
               // (if present) in the dynamic feature module are not loaded, but can be loaded by
-              // calling again with their loading unit id.
-              loadDartLibrary(
-                  sessionIdToLoadingUnitId.get(state.sessionId()),
-                  sessionIdToName.get(state.sessionId()));
-              sessionIdToName.delete(state.sessionId());
-              sessionIdToLoadingUnitId.delete(state.sessionId());
+              // calling again with their loading unit id. If no valid loadingUnitId was included in
+              // the installation request such as for an asset only feature, then we can skip this.
+              if (sessionIdToLoadingUnitId.get(sessionId) > 0) {
+                loadDartLibrary(
+                    sessionIdToLoadingUnitId.get(sessionId), sessionIdToName.get(sessionId));
+              }
+              if (channel != null) {
+                channel.completeInstallSuccess(sessionIdToName.get(sessionId));
```

Comment:
to make sure I understand, this is purely optional right? i.e. if a user calls loadLibrary in Dart (which ultimately leads here), they can just await that call instead of listening to this right?

Reply:
Yes, the methodChannel lives in parallel with the loadLibrary() code path. Since assets only modules do not have any library to load off of, this manual methodChannel way mirrors the functionality but allows specifying the moduleName directly. The channel accepts a loadingUnitId and a moduleName, but in the dart-side wrapper API I plan on exposing will restrict this to just taking a moduleName for now. The full loadingUnitId+moduleName API can be exposed in the future if/when dart allows initiating a loading unit load directly by loading unit id without a matching dart library loadLibrary() call.